ifacemonitor/ifacemonitor_test.go

```diff
@@ -63,26 +66,31 @@ func (nl *netlinkTest) addLink(name string) {
 		state: "down",
 		addrs: set.New(),
 	}
+	nl.linksMutex.Unlock()
```

Comment:
Should the mutex guard `nextIndex` too?  I'm not sure there's an issue but I'm having a hard time proving that to myself, whereas if it was guarded, it'd be easier.